github4s/src/main/scala/github4s/algebras/Repositories.scala

```diff
@@ -271,6 +271,26 @@ trait Repositories[F[_]] {
       headers: Map[String, String] = Map()
   ): F[GHResponse[List[User]]]
 
+  /**
+   * Get whether a user is a repository collaborator
+   *
+   * For organization-owned repositories, the list of collaborators includes outside collaborators,
+   * organization members that are direct collaborators, organization members with access through team memberships,
+   * organization members with access through default organization permissions, and organization owners.
+   *
```

Comment:
Might be worth mentioning that a 404 means the user is not a collaborator, 204 means user is a collaborator.

Reply:
or we might do the processing ourselves and return a `F[GHResponse[Boolean]]`

Reply:
I can change it to reflect this

Reply:
How about this?

* keep it as `GHResponse[Unit]`
* document the possible status codes (204, 404) and what they mean
* rename the method to `getCollaborator`

Reply:
I would expect a method by that name to return `GHResponse[User]`

Reply:
WDYT of the latest changes returning `F[GHResponse[Boolean]]`